src/battle_ai_util.c

```diff
@@ -404,7 +404,7 @@ bool32 BattlerHasAi(u32 battlerId)
 
 bool32 IsAiBattlerAware(u32 battlerId)
 {
-    if (AI_THINKING_STRUCT->aiFlags & AI_FLAG_OMNISCIENT)
+    if (AI_THINKING_STRUCT->aiFlags[battlerId] & AI_FLAG_OMNISCIENT)
```

Comment:
This breaks the `AI_FLAG_OMNISCIENT` flag. For example if you want to get the moves from `GetMovesArray` for the player, the id of the player mon will be passed which will be wrong because we need to check the if the AI is allowed to have access to players moves. 

I haven't checked others but it's possible that some others break as well.

Reply:
So I checked and as far as I can tell that's the only one that can get the wrong battler id. I don't know a clean fix though but wrote a small test to verify.
```
AI_SINGLE_BATTLE_TEST("AI should use a priority move if it is slower and faints to AI from next hit")
{
    GIVEN {
        AI_FLAGS(AI_FLAG_CHECK_VIABILITY | AI_FLAG_TRY_TO_FAINT | AI_FLAG_OMNISCIENT);
        PLAYER(SPECIES_WOBBUFFET) { Speed(20); Moves(MOVE_SLAM, MOVE_CELEBRATE); };
        OPPONENT(SPECIES_WOBBUFFET) { Speed(15); HP(1); Moves(MOVE_WATERFALL, MOVE_AQUA_JET); }
    } WHEN {
        TURN { MOVE(player, MOVE_SLAM); EXPECT_MOVE(opponent, MOVE_AQUA_JET); }
    }
}
```
AI is expected to use Aqua Jet but isn't even with the omniscient flag. If you replace the battle id with `B_POSITION_OPPONENT_LEFT` as a band-aid, it passes.